src/libexpr/eval.hh

```diff
@@ -310,7 +310,7 @@ private:
     friend struct ExprAttrs;
     friend struct ExprLet;
 
-    Expr * parse(const char * text, FileOrigin origin, const Path & path,
+    Expr * parse(char * text, size_t length, FileOrigin origin, const Path & path,
```

Comment:
It would be nicer to use a single `std::string_view` argument instead of separate `char *` and `size_t` arguments.

Reply:
we chose to split them to signify that *not* any `string_view` would be acceptable as the parser will fail if the last two bytes of the input aren't 0, but we've no hard feelings either way.

Reply:
upon closer inspection `string_view` will not work. the modified parser needs ownership of the input string so it can modify it, and taking a `string_view` to const_cast its data feels horrible :/